app/scripts/metamask-controller.js

```diff
@@ -1041,21 +1062,14 @@ export default class MetamaskController extends EventEmitter {
       SnapController: this.snapController,
       NotificationController: this.notificationController,
       ///: END:ONLY_INCLUDE_IN
+      ...resetOnRestartStore,
     });
 
```

Comment:
Suggestion: investigate whether we can add a test that will catch mistakes of adding stores to this.memstore but not this.store

Maybe this doesn't make sense because some memstores need to persist service worker restarts and some do not

Reply:
I'm not really sure how to go about this. :(